JavaShared/src/main/java/org/openzen/zenscript/javashared/JavaEnumMapper.java

```diff
@@ -0,0 +1,39 @@
+package org.openzen.zenscript.javashared;
+
+import org.openzen.zenscript.codemodel.definition.EnumDefinition;
+import org.openzen.zenscript.codemodel.member.EnumConstantMember;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class JavaEnumMapper {
+
+	private final Map<EnumDefinition, Set<EnumConstantMapping>> definitionMap;
+
+	public JavaEnumMapper() {
+		this.definitionMap = new HashMap<>();
+	}
+
+	public void merge(JavaEnumMapper mapper) {
+		mapper.definitionMap.forEach((key, val) -> this.definitionMap.merge(key, val, (a, b) -> Stream.concat(a.stream(), b.stream())
+				.collect(Collectors.toSet())));
+	}
+
+	public Optional<EnumConstantMapping> getMapping(EnumConstantMember member){
+
+		if(!(member.definition instanceof EnumDefinition)){
+			return Optional.empty();
+		}
+
+		return definitionMap.getOrDefault(member.definition, new HashSet<>()).stream().filter(mapping -> mapping.getMember().equals(member)).findFirst();
```

Comment:
`Collections.emptySet()`

At the same time, though, we're returning an optional anyway, so wouldn't it make more sense to create an optional directly and flatmap/map as needed, depending on whether we keep this structure or switch to nested maps?